settings/settingsTab.ts

```diff
@@ -0,0 +1,150 @@
+import {
+	PluginSettingTab,
+	Setting,
+    App
+} from "obsidian";
+
+import Blockreffer from "main";
+
+export class BlockrefferSettingTab extends PluginSettingTab {
+	plugin: Blockreffer;
+
+	constructor(app: App, plugin: Blockreffer) {
+		super(app, plugin);
+		this.plugin = plugin;
+	}
+
+	display(): void {
+		const {containerEl} = this;
+
+		containerEl.empty();
+
+		new Setting(containerEl)
+			.setHeading()
+			.setName("How do you want your link?");
+
+		new Setting(containerEl)
+			.setName('Link format')
+			.setDesc('How your link will be inserted into the document. Use {link} as a placeholder for the actual link.')
+			.addText(text => text
+				.setPlaceholder('!{link}')
+				.setValue(this.plugin.settings.format)
+				.onChange(async (value) => {
+					this.plugin.settings.format = value;
+					await this.plugin.saveSettings();
+				}));
+
+		new Setting(containerEl)
+			.setName("Use selected text as link display text")
+            .setDesc("If true, will use the text you have selected as the display text for the link (the bit that goes after the | symbol).")
```

Comment:
i like the character you added with "the bit" :sparkles: 

```suggestion
            .setDesc("Use selected text as the link alias (the bit that goes after the | symbol).")
```

Reply:
Hahaha thanks. I didn't know how else to explain it without it being overly complicated 😂